### PR TITLE
Comprehensive fileFormat aggregate (fixes #535)

### LINKED ISSUE #535: Incomplete list of total file formats and sizes returned in /summary response
Issue listed in HumanCellAtlas/data-browser#318

/summary does not return a complete list of file formats available. The response is limited by a secondary aggregate which returns only the top 10 results for the number of files available.

### PR DESCRIPTION
**Pull request by achave11**
Obtain fields for formation of fileFormat aggregate from one aggregate.
Added log for ES request of /summary endpoint.

### CODE REVIEW

File: src/azul/service/responseobjects/elastic_request_builder.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def create_query(filters):
         return Q('bool', must=query_list)
 
     @staticmethod
-    def create_aggregate(filters, facet_config, agg):
+    def create_aggregate(filters, facet_config, agg, request_config=None):
```
Comment (hannes-ucsc) [inline src/azul/service/responseobjects/elastic_request_builder.py]: I don't think the `request_config` argument can be optional.

File: test/service/test_response.py [hunk 2]
```diff
@@ -559,15 +559,15 @@ def test_file_search_response_add_facets(self):
...
-                self.assertGreater(summary_object['fileCount'], 0)
-                self.assertGreater(summary_object['organCount'], 0)
-                self.assertIsNotNone(summary_object['organSummaries'])
+        url = self.base_url + "repository/summary"
+        response = requests.get(url)
+        response.raise_for_status()
+        summary_object = response.json()
+        self.assertGreater(summary_object['fileCount'], 0)
+        self.assertGreater(summary_object['organCount'], 0)
+        self.assertGreater(len(summary_object['fileTypeSummaries']), 0)
```
Comment (hannes-ucsc) [inline test/service/test_response.py]: `assertGreaterOrEqual(…,1)` would be more expressive

File: test/service/test_response.py [hunk 3]
```diff
@@ -559,15 +559,15 @@ def test_file_search_response_add_facets(self):
...
+        self.assertGreater(summary_object['fileTypeSummaries'][0]['totalSize'], 0)
```
Comment (hannes-ucsc) [inline test/service/test_response.py]: same here

File: src/azul/service/responseobjects/elastic_request_builder.py [hunk 4]
```diff
@@ -101,9 +101,10 @@ def create_query(filters):
         return Q('bool', must=query_list)
 
     @staticmethod
-    def create_aggregate(filters, facet_config, agg):
+    def create_aggregate(filters, facet_config, agg, request_config):
         """
         Creates the aggregation to be used in ElasticSearch
+        :param request_config: Dictionary mappings for internal and external reference
```
Comment (hannes-ucsc) [inline src/azul/service/responseobjects/elastic_request_builder.py]: This sentence makes little sense to me. A dictionary IS a mapping. Not sure what you mean by "internal and external reference".

Comment (hannes-ucsc) [inline src/azul/service/responseobjects/elastic_request_builder.py]:
```suggestion
        :param request_config: a dictionary describing facets and field name mappings
```

File: CHANGELOG.yaml [hunk 5]
```diff
@@ -12,6 +12,13 @@ meta:
 
 changes:
 
+  - title: Adding a size_by_type field to the fileFormat aggregate
```
Comment (hannes-ucsc) [inline CHANGELOG.yaml]: Remember who your audience is. The changelog is for people outside of the Azul team: Mallory, Dave etc.

Your title describes a technical detail on how you solved this (and that description isn't very good either). Instead you need to describe how this affects clients of our service.

```suggestion
  - title: Fix: file type summary is truncated to ten entries
```

File: src/azul/service/responseobjects/elastic_request_builder.py [hunk 6]
```diff
@@ -407,6 +407,7 @@ def transform_summary(
         #  which has the format for the summary request
         self.logger.info('Creating a SummaryResponse object')
         final_response = SummaryResponse(es_response.to_dict())
+        logger.info("Elasticsearch request: %s", json.dumps(es_search.to_dict(), indent=4))
```
Comment (hannes-ucsc) [inline src/azul/service/responseobjects/elastic_request_builder.py]: I'd prefer this change in a separate commit.

File: CHANGELOG.yaml (hunk 5 above)
Comment (hannes-ucsc) [reply CHANGELOG.yaml]: You can reuse this changelog title as the commit title.

File: CHANGELOG.yaml [hunk 7]
```diff
@@ -12,6 +12,17 @@ meta:
 
 changes:
 
+  - title: Logger for ES request on /summary endpoint
```
Comment (hannes-ucsc) [inline CHANGELOG.yaml]: Remove this change log entry. The change log should only list changes that are observable by external users or that require an action by a team member when they upgrade their working copy to this version.

File: CHANGELOG.yaml [hunk 8]
```diff
@@ -12,6 +12,17 @@ meta:
...
+    upgrade:
+      - deploy
+
+  - title: Fix file type summary is truncated to ten entries
```
Comment (hannes-ucsc) [inline CHANGELOG.yaml]:
```suggestion
  - title: Fix: file type summary is truncated to ten entries
```

What's the reason for dropping the colon from my [previous](https://github.com/DataBiosphere/azul/pull/536#discussion_r233684382) suggestion?

Comment (achave11) [reply CHANGELOG.yaml]: Two test failed with the following error message:
```
yaml.scanner.ScannerError: mapping values are not allowed here
  in "/home/travis/build/DataBiosphere/azul/CHANGELOG.yaml", line 19, column 15
```

File: CHANGELOG.yaml (hunk 7 above)
Comment (achave11) [reply CHANGELOG.yaml]: @hannes-ucsc For the native test of change log, the failed test `test_changelog` specifies the following. 

```
    def test_changelog(self):
        self.run_setup('sdist', allow_fail=False)
        with open(os.path.join(self.package_dir, 'ChangeLog'), 'r') as f:
            body = f.read()
        # One commit, something should be in the ChangeLog list
        self.assertNotEqual(body, '')
``` 
And you specify the `logger` addition should be a different commit.

File: CHANGELOG.yaml (hunk 8 above)
Comment (achave11) [reply CHANGELOG.yaml]: @hannes-ucsc refer to [travis-logs](https://api.travis-ci.org/v3/job/455622756/log.txt)

**Pull request merged**